samples/subsys/smf/smf_calculator/src/main.c

```diff
@@ -0,0 +1,193 @@
+#include <zephyr/logging/log.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+LOG_MODULE_REGISTER(main_app);
+
+K_MSGQ_DEFINE(output_msgq, CALCULATOR_STRING_LENGTH, 2, 1);
+
+#define CALCULATOR_BUTTON_LABEL_LENGTH 4
+struct calculator_button {
+	char label[CALCULATOR_BUTTON_LABEL_LENGTH];
+	struct calculator_event event;
+};
+
+/**
+ * Important: this enum MUST be in the order of the buttons on the screen
+ * otherwise the mapping from button index to function will break
+ */
+enum calculator_ui_buttons {
+	BUTTON_CANCEL_ENTRY = 0,
+	BUTTON_CANCEL,
+	BUTTON_7,
+	BUTTON_8,
+	BUTTON_9,
+	BUTTON_DIVIDE,
+	BUTTON_4,
+	BUTTON_5,
+	BUTTON_6,
+	BUTTON_MULTIPLY,
+	BUTTON_1,
+	BUTTON_2,
+	BUTTON_3,
+	BUTTON_SUBTRACT,
+	BUTTON_0,
+	BUTTON_DECIMAL_POINT,
+	BUTTON_EQUALS,
+	BUTTON_ADD,
+};
+
+struct calculator_button buttons[] = {
+	[BUTTON_CANCEL_ENTRY] = {.label = "CE",
+				 .event = {.event_id = CANCEL_ENTRY, .operand = 'E'}},
+	[BUTTON_CANCEL] = {.label = "C", .event = {.event_id = CANCEL_BUTTON, .operand = 'C'}},
+	[BUTTON_ADD] = {.label = "+", .event = {.event_id = OPERATOR, .operand = '+'}},
+	[BUTTON_SUBTRACT] = {.label = "-", .event = {.event_id = OPERATOR, .operand = '-'}},
+	[BUTTON_MULTIPLY] = {.label = "*", .event = {.event_id = OPERATOR, .operand = '*'}},
+	[BUTTON_DIVIDE] = {.label = "/", .event = {.event_id = OPERATOR, .operand = '/'}},
+	[BUTTON_DECIMAL_POINT] = {.label = ".",
+				  .event = {.event_id = DECIMAL_POINT, .operand = '.'}},
+	[BUTTON_EQUALS] = {.label = "=", .event = {.event_id = EQUALS, .operand = '='}},
+	[BUTTON_0] = {.label = "0", .event = {.event_id = DIGIT_0, .operand = '0'}},
+	[BUTTON_1] = {.label = "1", .event = {.event_id = DIGIT_1_9, .operand = '1'}},
+	[BUTTON_2] = {.label = "2", .event = {.event_id = DIGIT_1_9, .operand = '2'}},
+	[BUTTON_3] = {.label = "3", .event = {.event_id = DIGIT_1_9, .operand = '3'}},
+	[BUTTON_4] = {.label = "4", .event = {.event_id = DIGIT_1_9, .operand = '4'}},
+	[BUTTON_5] = {.label = "5", .event = {.event_id = DIGIT_1_9, .operand = '5'}},
+	[BUTTON_6] = {.label = "6", .event = {.event_id = DIGIT_1_9, .operand = '6'}},
+	[BUTTON_7] = {.label = "7", .event = {.event_id = DIGIT_1_9, .operand = '7'}},
+	[BUTTON_8] = {.label = "8", .event = {.event_id = DIGIT_1_9, .operand = '8'}},
+	[BUTTON_9] = {.label = "9", .event = {.event_id = DIGIT_1_9, .operand = '9'}},
+};
+
+/* Where the result is printed */
+static lv_obj_t *result_label;
+
+/**
+ * LVGL v8.4 is not thread safe, so use a msgq to pass updates back
+ * to the thread that calls lv_task_handler()
+ */
+void update_display(const char *output)
+{
+	while (k_msgq_put(&output_msgq, output, K_NO_WAIT) != 0) {
+		k_msgq_purge(&output_msgq);
+	}
+}
+
+static void lv_btn_matrix_click_callback(lv_event_t *e)
+{
+	lv_event_code_t code = lv_event_get_code(e);
+	lv_obj_t *obj = lv_event_get_target(e);
+
+	if (code == LV_EVENT_PRESSED) {
+		uint32_t id;
+		int rc;
+
+		id = lv_btnmatrix_get_selected_btn(obj);
+		if (id > ARRAY_SIZE(buttons)) {
```

Comment:
i think the call from the line above returns 0-based index?

```suggestion
		if (id >= ARRAY_SIZE(buttons)) {
```